### ISSUE TITLE
Support external URI for contact avatar.

### ISSUE THREAD

**Issue by RokeJulianLockhart** [nextcloud/contacts#3979]
### Is your feature request related to a problem? Please describe.

I want my contact avatar to update according to the URI I've set in Libravatar, accessible at [`http://cdn.libravatar.org/avatar/ea13768f2a7107166add55082804a465?s=512`](http://cdn.libravatar.org/avatar/ea13768f2a7107166add55082804a465?s=512). [^1]

[^1]: [`dub.sh/s1hsiw`](https://dub.sh/s1hsiw)

### Describe the solution you'd like

Some vCard interpreters support URIs as avatar data, instead of encoded image data. [^4]

[^4]: [`stackoverflow.com/revisions/73311043/1`](https://stackoverflow.com/revisions/73311043/1#content:~:text=you%20need%20to%20include%20a%20base64%20encoded%20image%20URI)

### Describe alternatives you've considered

[`issues/2405`](https://github.com/nextcloud/contacts/issues/2405#issue-976567924) requests avatar lookup through Gravatar and Libravatar using an e-mail address as an identifier. [^2]  However, this is inferior: [^3]

[^2]: [`#issuecomment-2291005120`](https://github.com/nextcloud/contacts/issues/3979#issuecomment-2291005120)

[^3]: [`#issuecomment-2291011326`](https://github.com/nextcloud/contacts/issues/3979#issuecomment-2291011326)

> With a simple URI, both Gravatar, Libravatar, and any other image host is immediately supported in a standard manner. I see no necessity to integrate with a specific service using an identifier like an e-mail address.

**Comment by fazlerabbi37** [comment]
@RokeJulianLockhart possible duplicate of https://github.com/nextcloud/contacts/issues/2405?

**Comment by beedell-roke** [comment]
@fazlerabbi37, not quite. That issue appears to detail an API implementation with a specific service (albeit one which is self-hostable). Its scope, in that capacity, is somewhat unnecessarily restricted. If anything, it's more a dependency of this issue, depending upon its own implementation.

You see, with a simple URI, both Gravatar, Libravatar, and any other image host is immediately supported in a standard manner. I see no necessity to integrate with a specific service using an identifier like an e-mail address.

I've updated the issue to distinguish this from that issue. Thank you.

**Comment by hamza221** [comment]
This is supported by the rfc, so can be implemented. 
It's not a planned task at the moment but we welcome code contributions
